### PR TITLE
How to migrate an on-prem application to serverless?

### LINKED ISSUE #2968: How to migrate an on-prem application to serverless?
### Proposed title of article
How to migrate an on-prem application to serverless?

### Introduction paragraph:
Serverless is becoming popular in recent times. Developers and companies are adopting the approach to deliver business value using the public cloud. There are many articles covering the serverless architecture out there, but few really covers about the subject.
In this article, I will use a practical example, an Online Recruitment Portal which is hosted on-prem and cover the steps on how to migrate the web application to serverless.

### Key takeaways: 
At the end of the article, the reader will be able to understand:
- the problems faced with on-prem applications and why opt for seeverless way
- the journey to serverless solution and the migration approach from on-prem to serverless using practical example
- challenges likely to faced during development and delivery phase

### References: 
N?A

### PR DESCRIPTION
**Pull request by ogoti-lilian**
The article covers the following:
the problems faced with on-prem applications and why opt for seeverless way, the journey to serverless solution and the migration approach from on-prem to serverless using practical example, and the challenges likely to faced during development and delivery phase.

### Article checklist

**ATTENTION: In your PR - add the ISSUE (topic form) # that this PR corresponds to. You can do this by also adding a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.**
      
      
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**If you have not already please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve our overall technical writing.**

#### Preliminary Checks - Formatting and Structure 
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
This fixes #2968

### CODE REVIEW

**Comment by ogoti-lilian** [comment]
This solves issue #2968

**Comment by Ericgacoki** [comment]
Hello @ogoti-lilian, this article is well written and flows well. However, be sure to check out the [changes that I made](https://github.com/section-engineering-education/engineering-education/pull/3265/commits/d7402ddbc1de1e63c1991c7e537e5e5f237a50a1).

File: content/articles/how-to-migrate-an-on-prem-application-to-serverless/index.md [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+- **Integration**: Testing serverless applications is not accessible as one does not have access to the code's environment.
+
+- **Monitoring**: Serverless supports breaking down an application into smaller modules. This introduces a new challenge of distributed monitoring. With many serverless components bundled together, it is not easy to trace a request or response from end to end, and it is challenging to use with legacy monitoring tools. Although Azure Application Insights assists in monitoring Azure Functions, getting familiar with the metrics is difficult.
+
+### Conclusion
+It is a long journey to migrate from monolith architecture to serverless. It requires much investment and commitment from the management. Additionally, it also requires discipline from the team members that are developing it. However, the benefits associated with it can never be underestimated.
+
+It is usually a good idea to start small, rather than waiting for one big move. These incremental steps allow the developers to learn from their past mistakes, iterate, and try again. It is not a good idea to go for a perfect design. Instead, the team should be ready to iterate and improve.
```
Comment (Ericgacoki) [inline content/articles/how-to-migrate-an-on-prem-application-to-serverless/index.md:143]: Please paraphrase the first sentence.

**Review by Ericgacoki — CHANGES_REQUESTED** [review]
I have one suggestion at the moment...

**Comment by Ericgacoki** [comment]
@ogoti-lilian, resize your hero image to `600 X 400px`

**Comment by ogoti-lilian** [comment]
> Hello @ogoti-lilian, this article is well written and flows well. However, be sure to check out the [changes that I made](https://github.com/section-engineering-education/engineering-education/pull/3265/commits/d7402ddbc1de1e63c1991c7e537e5e5f237a50a1).

Thank you. I have noted the changes you made and I will improve on the next article.

**Comment by ogoti-lilian** [comment]
@Ericgacoki The changes requested have been made.
Thank you

**Review by Ericgacoki — APPROVED** [review]

**Comment by Ericgacoki** [comment]
Seems like you have decided to change your author image. Well, you may proceed.🤗

**Comment by ogoti-lilian** [comment]
> @ogoti-lilian, resize your hero image to `600 X 400px`

Sorry for the confusion the requested change has been made.
Thank you

**Comment by ogoti-lilian** [comment]
> Seems like you have decided to change your author image. Well, you may proceed.🤗

@Ericgacoki It was a slight mistake but I have rectified it. 
Thank you

**Comment by Ericgacoki** [comment]
No problem. It happens from time to time 😀
Remember to upload a better `hero` image :rocket:

**Comment by hectorkambow** [comment]
plagiarism check done

**Comment by Kayere** [comment]
Great article @ogoti-lilian - very informative and easy to follow through.
Thank you for the awesome review @Ericgacoki.

**Pull request merged**